### DISCUSSION TITLE
Multi vendor

### DISCUSSION THREAD

**Discussion by Milena92** [saleor/saleor#8599] [Q&A]
Hi, I've read online that some people implemented multi vendor store, using django-tenants, can anyone help me out with this?

**Comment by karolkielecki** [comment]
@Milena92 Hi Milena, afaik some community members were trying to implement it, but I'm not sure about the final result. Multi-vendor is a huge feature that would require changes everywhere. 

Regarding marketplace functionality -- there is only one official integration: [jetti.io/integrations/store/saleor](https://jetti.io/integrations/store/saleor).

**Reply by Milena92** [reply]
I haven't been able to find final solution either... Anyway, if anyone has done it, please let me know.

**Reply by karolkielecki** [reply]
@Milena92 As I mentioned earlier there is only one integration available. I'm afraid you can't build a multi-vendor using only django tenants. The whole API should be reworked - mutations, queries, etc.

**Reply by SpencerPatterson** [reply]
It seems to be a very popular request to have multi-vendor as part of Saleor. It was even on your todo list a year or two ago and then it was removed, why is that?

**Reply by karolkielecki** [reply]
Saleor hasn't been designed that way. We have other priorities. That's why there is an integration with Jetti. If someone would like to create a multivendor on top of Saleor -- then basically, the whole platform has to be reworked. It won't be easy to stay up to date with the next releases if you fork and modify the code to your needs.

**Comment by marqpdx** [comment]
What about exploiting the new channels aspect of saleor 3.0?
According to the docs:
```
Multiple channels can be managed through a single dashboard. Users can assign and manage the following unique data to specific channels:

* Currency. You can choose a unique currency for each channel. This function allows you to sell products in your customers’ currencies.

* Catalog and stock availability (products, collections). Users can set different products and collections for each channel. It allows you to manage two and more different sales points with different products and their availability.

* Discounts (vouchers and sales). Users can create and set multiple discounts for each channel.

* Shipping rates. Depending on the location of the store and shipping providers, you can set different shipping rates.

* Orders. Users can filter orders per channel. Draft orders can be created for each channel.
```

What seems to be missing is a way to sub-divide admin permissions, such that any given user can only work within one channel.
I haven't looked into it enough, but if all those features above are already in place, then we may be able to create a first level of multi-vendor within a single storefront by focusing on the authorization side of things.
Let me know if this makes any sense?

Thanks,
m

PS, just found this article on Django [multi-tenancy](https://bravelab.io/blog/how-to-manage-tenants-in-the-multitenant-app-based-on-django-tenants-and-saleor-io-platform/). Interesting read.

**Reply by karolkielecki** [reply]
I'm afraid exploiting the multichannel feature won't be enough. I believe multivendor would require much more work than you mentioned above. We have considered marketplace functionality in the past, but it is not something we plan to implement. But there is an integration with Jetti. I recommend checking that one: https://jetti.io/integrations/store/saleor
